### PR TITLE
Apply eslint-rule no-restricted-globals - Related to #1597

### LINKED ISSUE #1597: Enable pending eslint rules
### Expected behavior
Turned off `eslint` rules should be fixed to ensure code quality and to be able to use latest ECMA 6 features.
### Actual behavior
Bunch of `eslint` rules were turned off during the `eslint` integration to `1.0.0`, the mocha test specific eslint rules were fixed here  #1485. In root there are bunch of eslint rules still disabled, following are list of root eslint rules with number of errors.

## Manual fix
- [ ] "no-unused-expressions": 1252
- [ ] "camelcase": 813
- [ ] "no-param-reassign": 470
- [ ] "prefer-destructuring": 296
- [ ] "no-underscore-dangle": 291
- [ ] "no-plusplus": 154
- [ ] "max-len": 153
- [ ] "radix": 76
- [ ] "no-use-before-define": 76
- [ ] "no-new": 26

## Warnings
- [ ] "func-names": 1016

## DONE
- [x] "array-callback-return": #2563 
- [x] "callback-return": #2561 
- [x] "consistent-return": #2565 
- [x] "eqeqeq": #2562 
- ~"global-require": 69~ to keep until `app.js` is refactored
- [x] "guard-for-in": 3 #2648
- ~"generator-star-spacing": 2~ overwritten with Prettier
- [x] "new-cap": #2566
- [X] "no-bitwise": 13 #2590
- [X] "no-nested-ternary": 8 #2589
- [x] "no-prototype-builtins": 6 #2646
- [X] "no-restricted-globals": 9 #2649
- [x] "no-restricted-syntax": 7 #2643 
- [x] "no-shadow": 321 #2583
- ~"no-template-curly-in-string": 58~ to keep until https://github.com/LiskHQ/lisk/milestone/35 is accomplished
- ~"no-throw-literal": 29~ to keep
- [x] "prefer-rest-params": 5 #2640
- ~"strict": 286 #2585~
- [x] "vars-on-top": , 0 #2649
- ~"wrap-iife"~ overwritten with Prettier

## Auto fix using `npm run lint:fix`
- ~"space-before-function-paren": 2180~ https://github.com/prettier/prettier/issues/3847
- [X] "no-var": 4519 #2603

### Steps to reproduce
In root `.eslintrc.json` under `rules` section turn on the rules and run `npm run lint` to see the errors/warnings

### PR DESCRIPTION
**Pull request by diego-G**
### Review checklist

* The PR resolves #1597
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -64,7 +55,20 @@
 				]
 			}
 		],
+		"callback-return": "error",
+		"generator-star-spacing": [
```
[the remark this answers is not in the document]
Comment (diego-G) [reply .eslintrc.json]: The commit's title explains it. In a nutshell, I move to the bottom the eslint rules we are forced to keep or we want to keep.

[the remark this answers is not in the document]
Comment (diego-G) [reply .eslintrc.json]: In the same PR I've taken the freedom of removing another rule with 0 complains.

[the remark this answers is not in the document]
Comment (diego-G) [reply .eslintrc.json]: Please, leave this PR till the end.

**Comment by MaciejBaj** [comment]
The rule will be merged as the last rule from the #1597 scope (after https://github.com/LiskHQ/lisk/pull/2648 and https://github.com/LiskHQ/lisk/pull/2643).

**Pull request merged**